src/Raven.Server/Web/System/TcpConnectionInfoHandler.cs

```diff
@@ -19,5 +27,116 @@ public Task Get()
+        public async Task GetPullReplicationInfo()
+        {
+            var databaseName = GetStringQueryString("databaseName");
+            var databaseGroupId = GetStringQueryString("databaseGroupId");
+            var definitionName = GetStringQueryString("definitionName");
+
+            var nodes = GetResponsibleNode(databaseName, databaseGroupId, definitionName);
+
+            // TODO: cache the tcp info for each node?
+            DynamicJsonArray output = new DynamicJsonArray();
+            var clusterTopology = ServerStore.GetClusterTopology();
+            var tasks = new List<Task<TcpConnectionInfo>>();
+            foreach (var node in nodes)
+            {
+                if (node == ServerStore.NodeTag)
+                {
+                    output.Add(Server.ServerStore.GetTcpInfoAndCertificates(HttpContext.Request.GetClientRequestedNodeUrl()));
+                }
+                else
+                {
+                    var nodeUrl = clusterTopology.GetUrlFromTag(node);
+                    tasks.Add(GetNodeTcpInfo(nodeUrl, databaseName));
+                }
+            }
+
+            await Task.WhenAll(tasks);
+            foreach (var task in tasks)
+            {
+                output.Add(task.Result.ToJson());
+            }
+
+            using (ServerStore.ContextPool.AllocateOperationContext(out JsonOperationContext context))
+            using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+            {
+                context.Write(writer, new DynamicJsonValue
+                {
+                    ["Results"] = output
+                });
+            }
+        }
+
+        private async Task<TcpConnectionInfo> GetNodeTcpInfo(string nodeUrl, string databaseName)
+        {
+            using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext ctx))
+            using (var requestExecutor = ClusterRequestExecutor.CreateForSingleNode(nodeUrl, Server.Certificate.Certificate))
+            {
+                requestExecutor.DefaultTimeout = ServerStore.Engine.OperationTimeout;
+                var infoCmd = new GetTcpInfoCommand("database-group-tcp", databaseName);
+                await requestExecutor.ExecuteAsync(infoCmd, ctx);
+                return infoCmd.Result;
+            }
+        }
+
+        private List<string> GetResponsibleNode(string databaseName, string databaseGroupId, string definitionName)
+        {
+            var list = new List<string>();
+            using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                var pullReplication = ServerStore.Cluster.ReadPullReplicationDefinition(databaseName, definitionName, context);
+                var cert = HttpContext.Connection.ClientCertificate;
+                pullReplication.Validate(cert?.Thumbprint);
```

Comment:
This is wrong, because it presume too many states.
You need to account for: 
* http node
* https with a known certificate that doesn't have access to this db
* https with a known certificate that _has_ access to this db
* https with a known admin certificate
* https with an unknown certificate that is explicitly allowed, etc
* using a database that doesn't exists.

This gives you a good way to check (with no auth) if a db exists.